ssl/statem/statem_srvr.c

```diff
@@ -397,7 +396,8 @@ static WRITE_TRAN ossl_statem_server13_write_transition(SSL *s)
         return WRITE_TRAN_CONTINUE;
 
     case TLS_ST_SW_HELLO_RETRY_REQUEST:
-        return WRITE_TRAN_FINISHED;
+        st->hand_state = TLS_ST_EARLY_DATA;
+        return WRITE_TRAN_CONTINUE;
```

Comment:
This becomes CONTINUE instead of FINISHED so that we hit the post-work that flushes the state machine?  (Our write transition from TLS_ST_EARLY_DATA just returns WRITE_TRAN_FINISHED having done no work.)